src/hooks/useDrag.tsx

```diff
@@ -211,211 +236,208 @@ const updateHomepageState = (
         // safe to assert as check is done above
-      ] as AnnouncementsFrontmatterSection).announcements.announcement_items[
-        source.index
-      ]
-
-      const newAnnouncementsOptions = updatePositions(
-        (frontMatter.sections[
-          announcementsIndex
-          // safe to assert as check is done above
-        ] as AnnouncementsFrontmatterSection).announcements.announcement_items,
-        source.index,
-        destination.index,
-        draggedElem
-      )
-
-      const draggedError = errors.announcementItems[source.index]
-      const newAnnouncementErrors = updatePositions(
-        errors.announcementItems,
-        source.index,
-        destination.index,
-        draggedError
-      )
-      const displayBool = displayAnnouncementItems[source.index]
-      const newDisplayAnnouncementItems = updatePositions(
-        displayAnnouncementItems,
-        source.index,
-        destination.index,
-        displayBool
-      )
-
-      return updateAnnouncementSection(
-        homepageState,
-        newDisplayAnnouncementItems,
-        newAnnouncementsOptions,
-        newAnnouncementErrors,
-        announcementsIndex
-      )
-    }
-    case "textcardcard": {
-      const parentId = parseInt(typeArray[1], RADIX_PARSE_INT)
-      const textCardsItem = (frontMatter.sections[
-        parentId
-      ] as TextcardFrontmatterSection).textcards as EditorTextcardSection
-      const draggedElem = textCardsItem.cards[source.index]
-      const newTextcards = updatePositions(
-        textCardsItem.cards,
-        source.index,
-        destination.index,
-        draggedElem
-      )
-
-      const draggedError = errors.textcards[parentId][source.index]
-      const newTextcardErrors = updatePositions(
-        errors.textcards[parentId],
-        source.index,
-        destination.index,
-        draggedError
-      )
-
-      return updateTextCardsCardSection(
-        homepageState,
-        parentId,
-        newTextcards,
-        newTextcardErrors
-      )
-    }
-    default:
-      return homepageState
+      ] as AnnouncementsFrontmatterSection).announcements.announcement_items,
+      source.index,
+      destination.index,
+      draggedElem
+    )
+
+    const draggedError = errors.announcementItems[source.index]
+    const newAnnouncementErrors = updatePositions(
+      errors.announcementItems,
+      source.index,
+      destination.index,
+      draggedError
+    )
+    const displayBool = displayAnnouncementItems[source.index]
+    const newDisplayAnnouncementItems = updatePositions(
+      displayAnnouncementItems,
+      source.index,
+      destination.index,
+      displayBool
+    )
+
+    return updateAnnouncementSection(
+      homepageState,
+      newDisplayAnnouncementItems,
+      newAnnouncementsOptions,
+      newAnnouncementErrors,
+      announcementsIndex
+    )
   }
+  if (type.startsWith("textcardcard")) {
+    const parentId = parseInt(type.split("-")[1], RADIX_PARSE_INT)
```

Comment:
nit: add a comment here on why we can do this (cos it's been validated above to have the correct type so the second item is always a number)

Reply:
Added comments in https://github.com/isomerpages/isomercms-frontend/pull/1527/commits/96d0dddc2f7d06b599739083fad3887ffef41b9c